dotnet/src/Extensions/PromptTemplates.Liquid/LiquidPromptTemplate.cs

```diff
@@ -0,0 +1,252 @@
+        {
+            if (p.Default is not null)
+            {
+                this._inputVariables[p.Name] = p.Default;
+            }
+        }
+    }
+
+    /// <inheritdoc/>
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public async Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null, CancellationToken cancellationToken = default)
+#pragma warning restore CS1998
+    {
+        Verify.NotNull(kernel);
+        cancellationToken.ThrowIfCancellationRequested();
+        var variables = this.GetVariables(arguments);
+        var renderedResult = this._liquidTemplate.Render(variables);
+
+        // parse chat history
+        // for every text like below
+        // (system|assistant|user|function):
+        // xxxx
+        //
+        // turn it into
+        // <message role="system|assistant|user|function">
+        // xxxx
+        // </message>
+        var splits = s_roleRegex.Split(renderedResult);
+
+        // if no role is found, return the entire text
+        if (splits.Length > 1)
+        {
+            // otherwise, the split text chunks will be in the following format
+            // [0] = ""
+            // [1] = role information
+            // [2] = message content
+            // [3] = role information
+            // [4] = message content
+            // ...
+            // we will iterate through the array and create a new string with the following format
+            var sb = new StringBuilder();
+            for (var i = 1; i < splits.Length; i += 2)
+            {
+                var role = splits[i];
+                var content = splits[i + 1];
+                content = this.Encoding(content);
+                sb.Append("<message role=\"").Append(role).Append("\">").Append(LineEnding);
+                sb.Append(content).Append(LineEnding);
+                sb.Append("</message>").Append(LineEnding);
+            }
+
+            renderedResult = sb.ToString().TrimEnd();
+        }
+
+        return renderedResult;
+    }
+
+    private string Encoding(string text)
+    {
+        text = this.ReplaceReservedStringBackToColonIfNeeded(text);
+        text = HttpUtility.HtmlEncode(text);
+        return text;
+    }
+
+    private string ReplaceReservedStringBackToColonIfNeeded(string text)
+    {
+        if (this._allowUnsafeContent)
+        {
+            return text;
+        }
+
+        return text.Replace(ReservedString, ColonString);
+    }
+
+    /// <summary>
+    /// Gets the variables for the prompt template, including setting any default values from the prompt config.
+    /// </summary>
+    private Dictionary<string, object> GetVariables(KernelArguments? arguments)
+    {
+        var result = new Dictionary<string, object>();
+
+        foreach (var p in this._config.InputVariables)
+        {
+            if (p.Default == null || (p.Default is string stringDefault && stringDefault.Length == 0))
```

Comment:
@LittleLittleCloud, what's the reason for ignoring empty strings? Is that something we do in the other templates and it was copied here for consistency, or is it necessary for Liquid for some reason?

Reply:
It's copied from `HandlerBarTemplate.cs`, see here: https://github.com/microsoft/semantic-kernel/blob/431d18b93459673190d392f0e3b8faffbf417647/dotnet/src/Extensions/PromptTemplates.Handlebars/HandlebarsPromptTemplate.cs#L105

Reply:
@markwallace-microsoft, is this desirable? Mainly curious because of the inconsistency between default and provided args.

Reply:
I would have expected that setting the default value to be an empty string would work. I have created this issue https://github.com/microsoft/semantic-kernel/issues/6164 so we can follow up and make sure the behaviour is consistent across all of our prompt templates.